datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -90,6 +90,7 @@
   // lifecycle
   private volatile Thread _connectorTaskThread;
   protected volatile boolean _shutdown = false;
+  protected volatile boolean _failure = false;
```

Comment:
also, does this flag need to be `protected` or can it be changed to `private` ?

Reply:
We can make it private.